recipes/moc/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from os import rename
+from os.path import isdir, join
+from shutil import move
+from glob import glob
+
+class MocConan(ConanFile):
+    name = "moc"
+    homepage = "https://www.github.com/zuut/moc"
+    description = "Moc, the marked-up object compiler"
+    topics = ("conan", "moc", "idl", "code generation")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"    
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = { "fPIC": [True, False] }
+    default_options = { "fPIC": True }
+    _cmake = None
+    _supported_compilers = [
+        ("Linux", "gcc", "6"),
+        ("Linux", "clang", "6")
+    ]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self._check_compiler():
+            raise ConanInvalidConfiguration("%s package is not compatible with os %s and compiler %s version %s." % (self.name, self.settings.os, self.settings.compiler, self.settings.compiler.version))
+
+    def _check_compiler(self):
```

Comment:
We like to print a warning, you can use our template at conan-io/conan#8002

Reply:
It was throwing the ConanInvalidConfiguration because there were build failures in that case. NOTE: I figured out why it was failing. MOC expects to use /usr/lib/cpp c preprocessor which is there on the mac os systems I've used however its not on whatever macos build server machine that conan-center is using.